pkg/controllers/dynakube/injection/reconciler_test.go

```diff
@@ -444,6 +445,196 @@ func TestSetupEnrichmentInjection(t *testing.T) {
 	})
 }
 
+func TestGenerateCorrectInitSecret(t *testing.T) {
```

Comment:
There's a lot of code duplication in this tests. Would be nice if there's a way to factor parts of it to functions and re-use it (or alternatively do table tests ;-) )

Reply:
I agree, I previously tried to make it into a table test, but it became too convoluted for it's own good. So a scrapped it.
I will try to reduce the duplication.